simplex/docs/reconfiguration.md

```diff
@@ -0,0 +1,497 @@
+
+- The `observed_p_chain_height` is a deterministically encoded map that contains the P-chain heights that were observed by the nodes of the current epoch.
+  The key of the map is the node index that proposed the block, and the value is the P-chain height that was sampled by that node.
+  It's used to determine what would be the `next_p_chain_reference_height` of the next epoch, based on the P-chain heights sampled by the nodes so far.
+
+- The `sealing_block_seq` is the sequence number of the sealing block of the epoch, and it is set to `0` in all blocks that are not sealing blocks.
+  It is used to determine when the Simplex instance can transition to the next epoch.
+
+- The `block_validation_descriptor` describes how to validate the blocks of the next epoch.
+  It is only encoded in the sealing block of the epoch, and it is empty in all other blocks.
+  It is used to verify the finalization certificates of blocks that were built in the next epoch.
+
+- The `next_epoch_approvals` is a canoto message that contains the approvals of the next epoch by at least `n-f` nodes.
+  It is required in order to create the sealing block of the epoch and indicates that the nodes comprising the next epoch's validator set
+  are ready to start the next epoch. It prevents cases where nodes in the current epoch creating a sealing block and moving to the next epoch
+  while the nodes of the next epoch still replicating blocks and cannot yet participate in the consensus protocol.
+
+- The genesis block does not contain the Simplex epoch information, and it is implicitly set to be the zero values of the fields above.
+
+
+### Auxiliary information encoding
+
+Apart from the ICM epoch information and the Simplex epoch information, the block also contains auxiliary information
+that may be used by other protocols that require their information to be collected via the process of block building by the protocol participants.
+A common way for such information collection is to have each participant supply its input as part of the auxiliary information in the block, when it is its turn to build a block.
+
+The auxiliary information is encoded as a canoto message with the following schema:
+
+```proto
+message AuxiliaryInfo {
+   bytes info = 1; // The auxiliary information. Can be empty if no auxiliary information is provided.
+   uint64 prev_aux_info_seq = 2; // The sequence number of the previous block containing auxiliary information.
+   uint32 application_id = 3; // The application ID of the auxiliary information (if applicable).
+}
+```
+
+The `prev_aux_info_seq` field points to the previous block that contained auxiliary information, or `0` if the chain has no such block.
+As most blocks are not expected to contain auxiliary information, this provides a way for the application that uses the auxiliary information to
+efficiently find all blocks that contain auxiliary information without having to iterate over all blocks in the chain.
```

Comment:
We seem to have two different workarounds to avoid maintaining state:
1. Include O(n) duplicate information in the block header
2. Expect block verification to iterate backwards over O(n) blocks

Why don't we just maintain a current state? It seems more aligned with the rest of our implementations.